packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js

```diff
@@ -179,20 +173,17 @@ export function BlockSettingsDropdown( {
 		? undefined
 		: openedBlockSettingsMenu === currentClientId || false;
 
-	const onToggle = useCallback(
-		( localOpen ) => {
-			if ( localOpen && openedBlockSettingsMenu !== currentClientId ) {
-				setOpenedBlockSettingsMenu( currentClientId );
-			} else if (
-				! localOpen &&
-				openedBlockSettingsMenu &&
-				openedBlockSettingsMenu === currentClientId
-			) {
-				setOpenedBlockSettingsMenu( undefined );
-			}
-		},
-		[ currentClientId, openedBlockSettingsMenu, setOpenedBlockSettingsMenu ]
-	);
+	function onToggle( localOpen ) {
```

Comment:
While this is passed down, it's passed down to `useControlledValue()`:

https://github.com/WordPress/gutenberg/blob/01f02a07a71d08ac71cee66a6db4c0ea1d7ea9df/packages/components/src/dropdown/index.tsx#L69

which is then used for the initial value of `useState()`, thus is utilized only on the first `useState()` call anyway. 

Good to be un-memoized as well 👍